stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/connection.bal

```diff
@@ -9,3 +21,37 @@ public function <Connection conn> respondContinue () (HttpConnectorError) {
     HttpConnectorError err = conn.respond(res);
     return err;
 }
+
+@Description { value:"Sends a redirect response code to the user with given redirection status code." }
+@Param { value:"conn: The server connector connection" }
+@Param { value:"response: Response which user wants to send." }
+@Param { value:"redirectCode: Status code of the specific redirect." }
+@Param { value:"locations: Array of locations where the redirection can happen." }
+@Return { value:"Returns an HttpConnectorError if there was any issue in sending the response." }
+public function <Connection conn> redirect(OutResponse response, redirectCode code, string[] locations) (HttpConnectorError) {
+    if (code == redirectCode.MULTIPLE_CHOICES_300) {
+        response.statusCode = 300;
+    } else if (code == redirectCode.MOVED_PERMANENTLY_301) {
+        response.statusCode = 301;
+    } else if (code == redirectCode.FOUND_302) {
+        response.statusCode = 302;
+    } else if (code == redirectCode.SEE_OTHER_303) {
+        response.statusCode = 303;
+    } else if (code == redirectCode.NOT_MODIFIED_304) {
+        response.statusCode = 304;
+    } else if (code == redirectCode.USE_PROXY_305) {
+        response.statusCode = 305;
+    } else if (code == redirectCode.TEMPORARY_REDIRECT_307) {
+        response.statusCode = 307;
+    }
```

Comment:
Considering the lengthy `if-else` clauses required, does the use of enums for the status code seem justified? Wouldn't just integer constants be better suited? Then we can just get rid of the conditional checks.

Reply:
Yes. That is true. But my point is that there are only those given cases when redirecting. So giving an opportunity for the user to set any code is not appropriate. IMO either we use this way or give integer constant, we need to check whether the correct status code is set for the redirect.

Reply:
Well, you could have one conditional check to ensure the provided status code falls within the permissible values for redirect status codes.

Alternatively, if we had a way to add fields inside an enum (like in Java), this could be solved neatly using enums.